src/irmin/type.ml

```diff
@@ -583,16 +583,45 @@ end
 
 let compare = Compare.t
 
+
+exception Not_utf8
+
+let is_valid_utf8 str =
+  try
+    Uutf.String.fold_utf_8 (fun _ _ -> function
+        | `Malformed _ -> raise Not_utf8
+        | _ -> ()
+      ) () str;
+    true
+with Not_utf8 -> false
+
 module Encode_json = struct
 
   let lexeme e l = ignore (Jsonm.encode e (`Lexeme l))
 
   let unit e () = lexeme e `Null
 
-  (* what about escaping? *)
-  let string e s = lexeme e (`String s)
-  let bytes e s = lexeme e (`String (Bytes.unsafe_to_string s))
-  let char e c = string e (String.make 1 c)
+  let base64 e s =
+    let x = Base64.encode_exn s in
+    let () = lexeme e `Os in
+    let () = lexeme e (`Name "base64") in
+    let () = lexeme e (`String x) in
```

Comment:
Can you replace the `let () = X in` by `X;` ?

Reply:
No problem, just pushed an update to fix this